clients/spark/core/src/test/scala/io/treeverse/jpebble/BlockParserSpec.scala

```diff
@@ -261,3 +266,175 @@ class CountedIteratorSpec extends AnyFunSpec with Matchers {
     }
   }
 }
+
+class GolangContainerSpec extends AnyFunSpec with ForAllTestContainer {
+
+  override val container: GenericContainer = GenericContainer("golang:1.16.2-alpine",
+    classpathResourceMapping = Seq(
+      ("parser-test/generate-sst.sh", "/local/generate-sst.sh", BindMode.READ_WRITE),
+      ("parser-test/sst_files_generator.go", "/local/sst_files_generator.go", BindMode.READ_WRITE),
+      ("parser-test/go.mod", "/local/go.mod", BindMode.READ_WRITE),
+      ("parser-test/go.sum", "/local/go.sum", BindMode.READ_WRITE)),
+    command = Seq("/local/generate-sst.sh"),
+    waitStrategy = new LogMessageWaitStrategy().withRegEx("done\\n")
+  );
+
+  describe("A block parser") {
+    describe("with 2-level index sstable") {
+      it("should parse successfully") {
+        val fileName = "two.level.idx"
+        withGeneratedTestFiles(fileName, (in: BlockReadable, expected: Seq[(String, String)], sstSize: Long) => {
+          val it = BlockParser.entryIterator(in)
+          val actual = it.map((entry) =>
+            (new String(entry.key), new String(entry.value))).toSeq
+
+          actual should contain theSameElementsInOrderAs expected
+        })
+      }
+    }
+
+    val testFiles = Seq(
+      "fuzz.contents.0",
+      "fuzz.contents.1",
+      "fuzz.contents.2",
+      "fuzz.contents.3",
+      "fuzz.contents.4",
+      "fuzz.contents.5"
+    )
```

Comment:
I don't understand.  WHy are there 6 files?

In any case, maybe iterate over all the files in a directory if we can, since we have an application that generates them.